File: Chapter10/C#/RunningSamples/EncoderMedia/MediaPlayer/ScriptableTimelineMarker.cs
// <copyright file="ScriptableObservableCollection.cs" company="Microsoft">
//     Copyright © Microsoft Corporation. All rights reserved.
// </copyright>
// <summary>Implements a Scriptable version of the ObservableCollection class</summary>
// <author>Microsoft Expression Encoder Team</author>
using System.Windows;
using System.Windows.Browser;
using System.Windows.Media;

namespace ExpressionMediaPlayer
{
    [ScriptableType]
    public class ScriptableTimelineMarkerRoutedEventArgs : RoutedEventArgs
    {
        /// <summary>
        /// event args for when timeline marker reached
        /// </summary>
        /// <param name="marker"></param>
        public ScriptableTimelineMarkerRoutedEventArgs(ScriptableTimelineMarker marker)
        {
            Marker = marker;
        }

        public ScriptableTimelineMarker Marker { get; set; }
    }

    public delegate void ScriptableTimelineMarkerRoutedEventHandler(object sender, ScriptableTimelineMarkerRoutedEventArgs e);

    [ScriptableType]
    public class ScriptableTimelineMarker
    {
        /// <summary>
        /// text of marker
        /// </summary>
        public string Text { get; set; }
        /// <summary>
        /// time of marker (seconds)
        /// </summary>
        public double Time { get; set; }
        /// <summary>
        /// type of marker
        /// </summary>
        public string Type { get; set; }
        /// <summary>
        /// create new ScriptableTimelineMarker from TimelineMarker
        /// </summary>
        /// <param name="marker"></param>
        public ScriptableTimelineMarker(TimelineMarker marker)
        {
            Text = marker.Text;
            Time = marker.Time.TotalSeconds;
            Type = marker.Type;
        }
    }
}

File: Chapter10/VB/VBProject_10/AdvancedMedia/1-AdvanceMedia/AdvMediaPlayer.Web/EncoderMedia/MediaPlayer/ScriptableTimelineMarker.vb
'  <copyright file="ScriptableObservableCollection.cs" company="Microsoft">
'      Copyright © Microsoft Corporation. All rights reserved.
'  </copyright>
'  <summary>Implements a Scriptable version of the ObservableCollection class</summary>
'  <author>Microsoft Expression Encoder Team</author>
Imports System.Windows
Imports System.Windows.Browser
Imports System.Windows.Media

' Namespace ExpressionMediaPlayer

    <ScriptableType>
    Public Class ScriptableTimelineMarkerRoutedEventArgs
        Inherits RoutedEventArgs
        ''' <summary>
        ''' event args for when timeline marker reached
        ''' </summary>
        ''' <param name="marker"></param>
        Public Sub New(marker As ScriptableTimelineMarker)
            Marker = marker
        End Sub '   New


        Public Property Marker As ScriptableTimelineMarker
    End Class   '   ScriptableTimelineMarkerRoutedEventArgs


    public delegate Sub ScriptableTimelineMarkerRoutedEventHandler(sender As Object, e As ScriptableTimelineMarkerRoutedEventArgs)
    End Sub '   ScriptableTimelineMarkerRoutedEventHandler


    <ScriptableType>
    Public Class ScriptableTimelineMarker
        ''' <summary>
        ''' text of marker
        ''' </summary>
        Public Property Text As String
        ''' <summary>
        ''' time of marker (seconds)
        ''' </summary>
        Public Property Time As Double
        ''' <summary>
        ''' type of marker
        ''' </summary>
        Public Property Type As String
        ''' <summary>
        ''' create new ScriptableTimelineMarker from TimelineMarker
        ''' </summary>
        ''' <param name="marker"></param>
        Public Sub New(marker As TimelineMarker)
            Text = marker.Text
            Time = marker.Time.TotalSeconds

            Dim = As Type  marker.Type

        End Sub '   New
    End Class   '   ScriptableTimelineMarker
' End Namespace   '   ExpressionMediaPlayer
' ..\Project_10\AdvancedMedia\1-AdvanceMedia\AdvMediaPlayer.Web\EncoderMedia\MediaPlayer\ScriptableTimelineMarker.cs
